tls/s2n_server_hello.c

```diff
@@ -36,6 +37,52 @@
 /* From RFC5246 7.4.1.2. */
 #define S2N_TLS_COMPRESSION_METHOD_NULL 0
 
+/* From RFC8446 4.1.3. */
+#define S2N_DOWNGRADE_PROTECTION_SIZE   8
+const uint8_t tls12_downgrade_protection_bytes[] = {
+    0x44, 0x4F, 0x57, 0x4E, 0x47, 0x52, 0x44, 0x01
+};
+
+const uint8_t tls11_downgrade_protection_bytes[] = {
+    0x44, 0x4F, 0x57, 0x4E, 0x47, 0x52, 0x44, 0x00
+};
+
+static int s2n_detect_downgrade_mechanism(struct s2n_connection *conn) {
```

Comment:
this function could be guarded with s2n_is_tls13_enabled() too.

Reply:
/nit might be clearer to name it s2n_client_detect_downgrade_mechanism() vs s2n_server_add_downgrade_mechanism()

Reply:
Updated function names. The RFC says TLS1.2 client SHOULD check, so I didn't want to make it a tls13 only function.